bundles/framework/timeseries/service/TimeseriesService.js

```diff
@@ -0,0 +1,177 @@
+         */
+        registerTimeseries: function (id, type, priority, delegate, conf) {
+            conf = conf || {};
+            if (!id || !type || typeof priority !== 'number' || !delegate) {
+                throw new Error('All arguments must be given!');
+            }
+            var key = this._key(id, type);
+            this._timeseriesThings[key] = {
+                id: id,
+                type: type,
+                priority: priority,
+                delegate: delegate,
+                conf: conf
+            }
+            this._scheduleUpdate();
+        },
+        /**
+         * @method unregisterTimeseries
+         * Remove timeseries enabled thing (to remove UI from view)
+         * @param {String} id id of timeseries
+         * @param {String} type type of timeseries
+         */
+        unregisterTimeseries: function (id, type) {
+            if (!id || !type) {
+                throw new Error('All arguments must be given!');
+            }
+            var key = this._key(id, type);
+            var series = this._timeseriesThings[key];
+            if (series) {
+                delete this._timeseriesThings[key];
+                this._scheduleUpdate();
+            }
+            return series;
+        },
+        /**
+         * @method updateTimeseriesPriority
+         * Change priority of already registered timeseries
+         * @param {String} id id of target timeseries
+         * @param {String} type type of target timeseries
+         * @param {Number} priority new priority of target timeseries
+         */
+        updateTimeseriesPriority: function (id, type, priority) {
+            var key = this._key(id, type);
+            if (!this._timeseriesThings[key]) {
+                throw new Error('No timeseries found for type "' + type + '", id "' + id + '"!');
+            }
+            this._timeseriesThings[key].priority = priority;
+            this._scheduleUpdate();
+        },
+        /**
+         * @method getTimeseries
+         * Get previously registered timeseries
+         * @param {String} id id of timeseries
+         * @param {String} type type of timeseries
+         * @return {Object/undefined} timeseries config or undefined
+         */
+        getTimeseries: function (id, type) {
+            var key = this._key(id, type);
+            var thing = this._timeseriesThings[key];
+            return thing;
+        },
+        /**
+         * @method getActiveTimeseries
+         * Get timeseries that should have UI visible (has lowest priority)
+         * @return {Object/undefined} timeseries config or undefined
+         */
+        getActiveTimeseries: function () {
+            var me = this;
+            var things = Object.keys(this._timeseriesThings).map(function (key) { return me._timeseriesThings[key] });
+            var targets = things.filter(function (thing) {
+                return thing.type !== 'layer';
+            });
```

Comment:
Looks like map() and filter() are supported by the IE 9+ and modern browsers. We need to start using these in other places as well 👍